org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java

```diff
@@ -1765,4 +1387,20 @@ private boolean getConvertLambdaToMethodReferenceProposal(IInvocationContextCore
 		}
 		return false;
 	}
+
+	private boolean getAddMissingMethodDeclarationProposal(IInvocationContextCore context, ASTNode coveringNode, ArrayList<ProposalKindWrapper> resultingCollections) {
+		if (resultingCollections != null) {
+			var fix = AddMissingMethodDeclarationFixCore.createAddMissingMethodDeclaration(context.getASTRoot(), coveringNode);
+			if (fix != null) {
+				try {
+					var p = new ChangeCorrectionProposalCore(fix.getDisplayString(), fix.createChange(null), IProposalRelevance.ADD_INFERRED_LAMBDA_PARAMETER_TYPES);
+					resultingCollections.add(CodeActionHandler.wrap(p, CodeActionKind.QuickFix));

```

Comment:
@jjohnstn I'm not sure about this really, but since the original proposal was a QuickFix I kept it as it is.

Reply:
I think it's harmless, with one small but probably good-to-know difference :

https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/d0f09d3b886379ffe88d0993d63062dd682ed21f/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java#L312-L314

**CodeActionKind.QuickFix**
```json
{
        "title": "Add missing method 'foo' to class 'JSONStrings'",
        "kind": "quickfix",
        "diagnostics": [
            {
                "range": {
                    "start": {
                        "line": 19,
                        "character": 33
                    },
                    "end": {
                        "line": 19,
                        "character": 36
                    }
                },
                "severity": 1,
                "code": "67108979",
                "source": "Java",
                "message": "The method map(Function<? super Integer,? extends R>) in the type Stream<Integer> is not applicable for the arguments (JSONStrings::foo)"
            },
            {
                "range": {
                    "start": {
                        "line": 19,
                        "character": 37
                    },
                    "end": {
                        "line": 19,
                        "character": 53
                    }
                },
                "severity": 1,
                "code": "603979903",
                "source": "Java",
                "message": "The type JSONStrings does not define foo(Integer) that is applicable here"
            }
        ],
        "data": {
            "pid": "1",
            "rid": "2"
        }
    }
```

**JavaCodeActionKind.QUICK_ASSIST**
```json
{
        "title": "Add missing method 'foo' to class 'JSONStrings'",
        "kind": "quickassist",
        "data": {
            "pid": "2",
            "rid": "2"
        }
    }
```


It's probably a bug because a quick assist shouldn't need diagnostic data. I don't think it's harmful to set the kind of a code action to `CodeActionKind.QuickFix`, although I notice a lot of other references in